molo/forms/tests/test_reaction_questions.py

```diff
@@ -0,0 +1,554 @@
+        choice1 = question.get_children().first()
+        self.assertEqual(ReactionQuestionResponse.objects.all().count(), 0)
+
+        kw = {'question_id': question.id, 'article_slug': article.slug}
+        data = {'choice': choice1.id}
+        response = self.client.post(
+            reverse('molo.forms:reaction-vote', kwargs=kw), data=data)
+
+        self.assertEqual(ReactionQuestionResponse.objects.all().count(), 1)
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(
+            response['Location'],
+            '/forms/reaction/test-page-0/{}/yes/feedback/'.format(question.pk)
+        )
+
+        response = self.client.get(
+            '/forms/reaction/test-page-0/{}/yes/feedback/'.format(question.pk)
+        )
+
+        self.assertContains(
+            response, '<a href="/sections-main-1/your-mind/test-page-0/">')
+
+        self.assertContains(response, 'well done')
+
+        # test user can only submit once
+        response = self.client.post(
+            reverse('molo.forms:reaction-vote', kwargs=kw),
+            data=data)
+        self.assertEqual(ReactionQuestionResponse.objects.all().count(), 1)
+        self.assertEqual(
+            ReactionQuestionResponse.objects.last().choice.pk, choice1.pk)
+
+        messages = list(get_messages(response.wsgi_request))
+        self.assertEqual(len(messages), 1)
+        # this should show if the submit is not done via ajax
+        self.assertIn(
+            'You have already given feedback on this article.',
+            str(messages[0])
+        )
+
+    def test_can_react_on_article_ajax_call(self):
+        article = self.mk_article(self.yourmind)
+        article.save_revision().publish()
+        question = self.mk_reaction_question(self.reaction_index, article)
+        self.user = self.login()
+
+        kw = {'question_id': question.id, 'article_slug': article.slug}
+        response = self.client.get(article.url)
+
+        self.assertContains(response, question.title)
+        for choice in question.get_children():
+            self.assertContains(response, choice.title)
+
+        choice2 = question.get_children().last()
+        # submit with ajax in post data
+        response = self.client.post(
+            reverse('molo.forms:reaction-vote', kwargs=kw),
+            {'choice': choice2.id, 'ajax': 'True'})
+
+        # there should still only be one response object from user
+        # but the vote should change from choice 1 to choice 2
+        self.assertEqual(ReactionQuestionResponse.objects.all().count(), 1)
+        self.assertEqual(
+            ReactionQuestionResponse.objects.last().choice.pk, choice2.pk)
+        messages = list(get_messages(response.wsgi_request))
+        # no error message if the submit is done via ajax
+        self.assertEqual(len(messages), 0)
+
+        response = self.client.post(
+            reverse('molo.forms:reaction-vote', kwargs=kw),
+            {'choice': choice2.id, 'ajax': 'True'})
```

Comment:
Please change the choice here. We're trying to test that if the user changes their choice it doesn't make a new response object but overwrites the old one